### ISSUE TITLE
Smarty_Internal_ErrorHandler doesn't check error_reporting() to comply with php error configuration

### ISSUE THREAD

**Issue by eposjk** [smarty-php/smarty#768]
When I set error_reporting in my application to suppress e.g. E_NOTICE (or other) errors, smarty 4.x does not respect that setting when it calls some of my custom smarty plugins.

The [documentation of set_error_handler](https://www.php.net/manual/function.set-error-handler.php) mentions that custom error handlers are called independent of the configured error levels, and contains example code how to respect the configured error levels in a custom error handler.

Applied to smarty, the solution for that problem should be to add at the beginning of

https://github.com/smarty-php/smarty/blob/f8e1ba76f77e17ec7b364b2922d350990ead7168/libs/sysplugins/smarty_internal_errorhandler.php#L67-L78
the following lines of code:
```
        if (!(error_reporting() & $errno)) {
            return; // suppresses this error
        }
```

This might also solve some of the other issues related to error handling ( #759 and maybe others )

**Comment by wisskid** [comment]
@eposjk what exactly is the problem that you are experiencing? 

Currently, `\Smarty_Internal_ErrorHandler::handleError` only silences some errors. The problem is probably not the `error_reporting` setting, but the following: if you registered your previous custom error handler with a value for the second parameter to `set_error_handler`, Smarty will call it regardless, as it cannot retrieve this value. So if your previous custom error handler does not check the error level itself, you may end up with unexpected results.

Ideally, we should not register a custom error handler in Smarty, or only temporarily and restore it afterwards with `restore_error_handler()`, but that might mean that previously error handlers wouldn't work at all when rendering a template. Probably not great either.

**Comment by eposjk** [comment]
You're right.
When I added this fix half a year ago to my own codebase, it seemed to help to suppress unwanted php warningsI was fighting PHP warnings. Later I added some more fixes for php warnings in my own error handler. Now i tried for three hours to reproduce the difference of (not) checking error_reporting in handleError(), but I couldn't. Your explanation makes sense - and as you don't do any actions in handleError(), it makes sense that checking error_reporting in my own error handler is enough.
Thanks for your great work with Smarty, and sorry for all circumstances.
